### PR TITLE
Update Python minor toolchain versions to allow smaller Python binaries

### PR DESCRIPTION
**Pull request by tetsuok**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [x] Other... Please describe:

Updates the Python toolchain versions for 3.8.13, 3.9, and 3.10. Adds 3.9.13 and 3.10.6. The `install_only` archives of 3.8.13, 3.9.13, and 3.10.6 are smaller than before, because the static libpython libraries and tests for the stdlib itself are removed from the archives. This is essentially equivalent to what we do in #758, but now in the python-build-standalone side. We can remove our own filtering introduced in #758 once the support of earlier versions than 3.8.13, 3.9.13, and 3.10.6 is dropped.
For details of the toolchain side changes, see https://github.com/indygreg/python-build-standalone/releases/tag/20220802.

## What is the current behavior?

Issue Number: N/A

For 3.9, 3.9.12 is installed.
For 3.10, 3.10.4 is installed.

## What is the new behavior?

For 3.9, 3.9.13 is installed.
For 3.10, 3.10.6 is installed.

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

When user sets `python_version = "3.9"` in `python_register_toolchains`, 3.9.13 is installed.
When user sets `python_version = "3.10"` in `python_register_toolchains`, 3.10.6 is installed.

### CODE REVIEW

File: python/versions.bzl [hunk 1]
```diff
@@ -102,13 +113,24 @@ TOOL_VERSIONS = {
...
+    },
 }
 
 # buildifier: disable=unsorted-dict-items
 MINOR_MAPPING = {
     "3.8": "3.8.13",
-    "3.9": "3.9.12",
-    "3.10": "3.10.4",
+    "3.9": "3.9.13",
+    "3.10": "3.10.6",
```
Comment (mattem) [inline python/versions.bzl:133]: I guess this PR is a superset of https://github.com/bazelbuild/rules_python/pull/785. I'm not sure I agree with changing the minor mappings and silently bumping users versions. Thinking about it again though they explicitly set the version with no patch, so maybe it is okay...?

Comment (tetsuok) [reply python/versions.bzl:133]: @mattem I changed this by following the previous similar PR #707.

Comment (tetsuok) [reply python/versions.bzl:133]: I'm totally fine to remove this minor mapping change if separate discussion is needed on when to / how to update minor mappings.

Comment (mattem) [reply python/versions.bzl:133]: Ah okay, we've set a precedence that these get updated previously. I think we are okay here then 👍

**Review by mattem — APPROVED** [review]

File: python/versions.bzl (hunk 1 above)
Comment (tetsuok) [reply python/versions.bzl:133]: Sure. Thanks for the review!

**Pull request merged**